plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/build/ProjectBuilder.java

```diff
@@ -298,13 +297,10 @@ protected XBuildResult doBuild(
 	/** Deletes the contents of the output directory */
 	public void doClean(AfterDeleteListener deleteListener, CancelIndicator cancelIndicator) {
 		deletePersistenceFile();
+		doClearWithNotification();
```

Comment:
It would be more consistent to clear issues only if indexOnly == false, because in case we are not responsible for performing validation and thus not responsible for _creating_ issues, we should not be responsible for _deleting_ issues. However, in case of issues it is probably better to ensure a clean slate.

Maybe add a comment along the lines of "clear issues on client side even in indexOnly mode, because ...".